diff --git a/src/parse.rs b/src/parse.rs
index e92d87a..f7edc5a 100644
--- a/src/parse.rs
+++ b/src/parse.rs
@@ -269,7 +269,7 @@ fn op(input: &str) -> (Op, &str) {
     if bytes.first() == Some(&b'=') {
         (Op::Exact, &input[1..])
     } else if bytes.first() == Some(&b'>') {
-        if bytes.get(1) == Some(&b'=') {
+        if Some(&b'=') == bytes.get(1) {
             (Op::GreaterEq, &input[2..])
         } else {
             (Op::Greater, &input[1..])
