diff --git a/serde_core/src/ser/impls.rs b/serde_core/src/ser/impls.rs
index a7a175d..e230cd4 100644
--- a/serde_core/src/ser/impls.rs
+++ b/serde_core/src/ser/impls.rs
@@ -774,7 +774,7 @@ fn format_u8(mut n: u8, out: &mut [u8]) -> usize {
         out[1] = DEC_DIGITS_LUT[d1];
         out[2] = DEC_DIGITS_LUT[d1 + 1];
         3
-    } else if n >= 10 {
+    } else if n > 9 {
         let d1 = (n << 1) as usize;
         out[0] = DEC_DIGITS_LUT[d1];
         out[1] = DEC_DIGITS_LUT[d1 + 1];
