diff --git a/src/eval.rs b/src/eval.rs
index a6e30e2..c2a5e82 100644
--- a/src/eval.rs
+++ b/src/eval.rs
@@ -69,7 +69,7 @@ fn matches_greater(cmp: &Comparator, ver: &Version) -> bool {
 
     match cmp.patch {
         None => return false,
         Some(patch) => {
-            if ver.patch != patch {
+            if ver.patch >= patch {
                 return ver.patch > patch;
             }
         }
